Compilation failure on MacOS Sonoma

Gem version 0.17.1 fails to compile on MacOS Sonoma:
```
% gem install prism
Fetching prism-0.17.1.gem
Building native extensions. This could take a while...
ERROR:  Error installing prism:
	ERROR: Failed to build gem native extension.

    current directory: /Users/jonathan/.gem/ruby/3.2.2/gems/prism-0.17.1/ext/prism
/Users/jonathan/.rubies/ruby-3.2.2/bin/ruby extconf.rb
checking for whether -fvisibility=hidden is accepted as CFLAGS... yes
compiling build/static/diagnostic.o
compiling build/static/token_type.o
compiling build/static/prettyprint.o
compiling build/static/util/pm_strpbrk.o
compiling build/static/util/pm_char.o
compiling build/static/util/pm_memchr.o
compiling build/static/util/pm_list.o
compiling build/static/util/pm_strncasecmp.o
compiling build/static/util/pm_state_stack.o
compiling build/static/util/pm_string.o
compiling build/static/util/pm_newline_list.o
compiling build/static/util/pm_string_list.o
compiling build/static/util/pm_buffer.o
compiling build/static/util/pm_constant_pool.o
compiling build/static/serialize.o
serialize.c.erb:290:30: error: suggest braces around initialization of subobject [-Werror,-Wmissing-braces]
    pm_options_t options = { 0 };
                             ^
                             {}
serialize.c.erb:325:30: error: suggest braces around initialization of subobject [-Werror,-Wmissing-braces]
    pm_options_t options = { 0 };
                             ^
                             {}
2 errors generated.
make: *** [build/static/serialize.o] Error 1
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/Users/jonathan/.rubies/ruby-3.2.2/bin/$(RUBY_BASE_NAME)
	--enable-debug-mode-build
	--disable-debug-mode-build
	--enable-static
	--disable-static
extconf.rb:78:in `system': Command failed with exit 2: make (RuntimeError)
	from extconf.rb:78:in `block in build_target_rubyparser'
	from extconf.rb:72:in `chdir'
	from extconf.rb:72:in `build_target_rubyparser'
	from extconf.rb:68:in `build_static_rubyparser'
	from extconf.rb:42:in `configure_rubyparser'
	from extconf.rb:20:in `configure'
	from extconf.rb:136:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /Users/jonathan/.gem/ruby/3.2.2/extensions/x86_64-darwin-23/3.2.0-static/prism-0.17.1/mkmf.log

extconf failed, exit code 1
```

Gem version 0.16.0 compiles successfully on MacOS Sonoma:
```
% gem install prism --version 0.16.0
Fetching prism-0.16.0.gem
Building native extensions. This could take a while...
Successfully installed prism-0.16.0
Parsing documentation for prism-0.16.0
Installing ri documentation for prism-0.16.0
Done installing documentation for prism after 2 seconds
1 gem installed
```

--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@ FUZZ_OUTPUT_DIR = $(shell pwd)/fuzz/output
 SOEXT := $(shell ruby -e 'puts RbConfig::CONFIG["SOEXT"]')
 
 CPPFLAGS := -Iinclude
-CFLAGS := -g -O2 -std=c99 -Wall -Werror -Wextra -Wpedantic -Wundef -Wconversion -fPIC -fvisibility=hidden
+CFLAGS := -g -O2 -std=c99 -Wall -Werror -Wextra -Wpedantic -Wundef -Wconversion -Wno-missing-braces -fPIC -fvisibility=hidden
 CC := cc
 WASI_SDK_PATH := /opt/wasi-sdk
 

--- a/docs/building.md
+++ b/docs/building.md
@@ -10,7 +10,7 @@ All of the source files match `src/**/*.c` and all of the headers match `include
 The following flags should be used to compile prism:
 
 * `-std=c99` - Use the C99 standard
-* `-Wall -Wconversion -Wextra -Wpedantic -Wundef` - Enable the warnings we care about
+* `-Wall -Wconversion -Wextra -Wpedantic -Wundef -Wno-missing-braces` - Enable the warnings we care about
 * `-Werror` - Treat warnings as errors
 * `-fvisibility=hidden` - Hide all symbols by default
 

